app/assets/javascripts/contacts.js

```diff
@@ -0,0 +1,111 @@
+var accepts = "application/vnd.addressbook_service+json; version=1";
+var async = true;
+var contentType = "application/json";
+var url = "http://localhost:5100/api/contacts";
+var storedContacts = {};
+
+var erredContacts = function(_xhr, status, error) {
+  // FIXME: Display result to the user
+  console.log("retrieveContacts was unsuccessful");
+  console.log("Status: " + status);
+  console.log("Error:" + error);
+};
+
+var erredUpdate = function(_xhr, status, error) {
+  // FIXME: Display result to the user
+  console.log("updateContacts was unsuccessful");
+  console.log("Status: " + status);
+  console.log("Error:" + error);
+};
+
+var processContacts = function(contacts, status, xhr) {
+  storedContacts = contacts;
+
+  for (var i = 0; i < storedContacts.length; i++) {
+    $(".contacts").append(
+      "<li><a data-role='contact' data-value='" + i + "' href='#'>" +
+      storedContacts[i].name + "</a></li>"
+    );
+  }
+};
+
+var showContactInForm = function(position) {
+  var contact = storedContacts[position];
+
+  $("[data-role='contact-header']").html("Contact " + contact.id);
+  $("[data-role='id']").val(contact.id);
+  $("[data-role='email']").val(contact.email);
+  $("[data-role='phone']").val(contact.phone);
+  $("[data-role='name']").val(contact.name);
+  $("[data-role='notes']").val(contact.notes);
+};
+
+var clickContacts = function() {
+  $("a[data-role='contact']").click( function() {
+    showContactInForm($(this).data("value"));
+  });
+};
+
+var showSuccess = function() {
+  // FIXME: Display result to the user
+  console.log("Form submission worked!");
+};
+
+var clearContactForm = function() {
```

Comment:
Not used in this file, it is used from application.js and bundled as part of the asset pipeline.